Lib/test/test_importlib/util.py

```diff
@@ -298,7 +298,7 @@ def writes_bytecode_files(fxn):
     """Decorator to protect sys.dont_write_bytecode from mutation and to skip
     tests that require it to be set to False."""
     if sys.dont_write_bytecode:
-        return lambda *args, **kwargs: None
+        return unittest.skip(fxn)
```

Comment:
```suggestion
        return unittest.skip("relies on writing bytecode")(fxn)
```
Similar to `Lib/test/support/socket_helper.py:        return unittest.skip(msg)(test)`.